client/src/AppBundle/Resources/views/Report/Report/overview.html.twig

```diff
@@ -290,31 +275,58 @@
     </ul>
 
     {% if not report.isDue %}
-
         <div class="push--top">
             {% set endDate = report.getEndDate() | date("j F Y")  %}
             {% set transOptions = {'%date%': endDate} %}
 
             {{ macros.notification('info', 'previewNoticeNotDue' | trans(transOptions)) }}
         </div>
-        <a id="edit-report-preview" class="button behat-link-edit-report-preview" href="{{ path('report_review', {'reportId': report.id}) }}">{{ 'options.previewReport' | trans() }}</a>
-
     {% else %}
-        {% if not reportStatus.isReadyToSubmit %}
-
-            <div class="push--top">
-                {{ macros.notification('info', 'previewNotice' | trans()) }}
-            </div>
-            <a id="edit-report-preview" class="govuk-button behat-link-edit-report-preview" href="{{ path('report_review', {'reportId': report.id}) }}" data-module="govuk-button">{{ 'options.previewReport' | trans() }}</a>
-
-        {% else %}
-
-            <a id="edit-report-review" class="govuk-button push--top behat-link-edit-report-review" href="{{ path('report_review', {'reportId': report.id}) }}" data-module="govuk-button">{{ 'continue' | trans({}, 'common') }}</a>
-
-        {% endif %}
+        <div id="js-enabled-controls">
+            {% if report.isUnsubmitted %}
+                <div class="govuk-form-group">
+                    <fieldset class="govuk-fieldset">
+                        <div class="govuk-checkboxes">
+                            <div class="govuk-checkboxes__item">
+                                <input class="govuk-checkboxes__input" name="confirmReview" type="checkbox" data-module="opg-button-toggler">
+                                <label class="govuk-label govuk-checkboxes__label" for="confirmReview">
+                                    {{ 'confirmSectionsReviewed' | trans() }}
+                                </label>
+                            </div>
+                        </div>
+                    </fieldset>
+                </div>
+            {% endif %}
+            <a id="{{ report.status.isReadyToSubmit ? 'edit-report-review' : 'edit-report-preview' }}"
+               class="push--top behat-link-edit-report-review"
+               href="{{ path('report_review', {'reportId': report.id}) }}"
+            >
+                <button
+                    class="govuk-button {{ report.isUnsubmitted ? 'govuk-button--disabled' : '' }}"
+                    aria-disabled="{{ report.isUnsubmitted ? 'true' : 'false' }}"
+                    {% if report.isUnsubmitted %}
+                        disabled="disabled"
+                    {% endif %}
+                    data-module="opg-toggleable-button"
+                >
+                    {{ 'options.previewReport' | trans() }}
+                </button>
+            </a>
+        </div>
 
+        {# Fallback for JS disabled browsers #}
+        <noscript>
```

Comment:
An alternative for this is using `js-hidden` which is automatically hidden if JavaScript is enabled.
```suggestion
        <div class="js-hidden">
```
We don't currently have a reverse class for the other div, but could easily create `js-only` with the CSS:
```
.js-only { display: none; }
.js-enabled .js-only { display: block; }
```

Reply:
Thats what I wanted - just couldn't figure out how to apply it. Cheers, will update.

Reply:
Aaaand this magical line is how it works `<script>document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');</script>`. OK cool, totally getting this now. Much nicer solution.